### PR TITLE
Call Dispose after Xunit test where test class inherits IDisposable

### PR DESCRIPTION
**Pull request by LaurenceJKing**
#514 

This pull request aims to resolve the bug uncovered in issue 514 (Dispose not called from IDisposable class when using xunit).

### CODE REVIEW

**Comment by ploeh** [comment]
As in #514, I'm still concerned that this only disposes _once_, while each property runs (by default) 100 times.

What problem does this address?

I'm not dead set against this change, but I think that maintainers (which is mostly @kurtschelfthout, so my voice shouldn't count for much) should be conservative when it comes to adding behaviour that may not be well understood. There's always a risk of painting oneself into a corner... I can't tell if there's a risk of that here.

File: src/FsCheck.Xunit/PropertyAttribute.fs [hunk 1]
```diff
@@ -205,6 +213,7 @@ type PropertyTestCase(diagnosticMessageSink:IMessageSink, defaultMethodDisplay:T
 
                     match xunitRunner.Result with
                           | TestResult.True _ ->
+                            dispose target
```
Comment (ploeh) [inline src/FsCheck.Xunit/PropertyAttribute.fs:216]: Do understand this correctly, that the code only disposes the `target` if the result is `True`?

Isn't the desired behaviour that the object should always be disposed, regardless of outcome?

Comment (LaurenceJKing) [reply src/FsCheck.Xunit/PropertyAttribute.fs:216]: My mistake. I thought xunit only called dispose after a successful test, but it turns out I was wrong. I'll amend amend this to always dispose

**Comment by ploeh** [comment]
Apart from my concerns, _if_ we want this behaviour, I think that the pull request looks fine.

**Comment by kurtschelfthout** [comment]
@ploeh Thanks very much for your considered comments. As I see it, this PR addresses a _bug_ with how `PropertyAttribute` is wired into xUnit - I consider it a bug because if you have a disposable class with a `FactAttribute` method in it, `Dispose` will be called (once). Replace the `FactAttribute` with `PropertyAttribute` and now suddenly `Dispose` is no longer called. That is counter-intuitive. 

The ability to clean up after every FsCheck test is an interesting feature request, but is I believe orthogonal, and I wouldn't want FsCheck to call `Dispose` after every test. That would be similar to doing for a plain old object:

```fsharp
let o = new MyObject()
for i in [1..100] do
    o.MyMethodCall()
    o.Dispose()
```
which seems weird to me - `Dispose` is to `new` like `free` is to `malloc` in less GC-englightened languages ;)

I hope that makes sense.

**Pull request merged**